### PR TITLE
WIP: [ENTESB-14902] Fix typo and use streams more

### PR DESCRIPTION
**Pull request by VratislavHais**

### CODE REVIEW

**Comment by VratislavHais** [comment]
Code should be equal. It needs to be tested, with this I will wait for you. One of the advantages of this "streams" approach is that it can be easily changed to parallelStream() to boost performance a bit. Also it might be a bit quicker. Not sure about this though.

**Comment by grgrzybek** [comment]
@VratislavHais thanks for the PR, but I have to read it more carefully as I'm usually reluctant to stream/lambda heavy code ;)

**Comment by grgrzybek** [comment]
For example [this](https://github.com/apache/aries-jax-rs-whiteboard/blob/master/jax-rs.whiteboard/src/main/java/org/apache/aries/jax/rs/whiteboard/internal/Whiteboard.java#L1101-L1134) is totally unreadable for me ;)

**Comment by VratislavHais** [comment]
uf, this is some hardcore stuff @grgrzybek ! Anyway my lamdas should be identical to your code. Except the most complicated one:
```
final Map<Boolean, List<String>> partitioned = added.keySet().stream().collect(Collectors.partitioningBy(name -> !deployedPatches.containsKey(name)));
            partitioned.get(true).forEach(name -> deployedPatches.put(name, new DeployedPatch(added.get(name), 0L)));
```
this one is not identical because it made the lambda much easier to just ignore values in the map. I have noticed from your code that values are not used anywhere later in the code so I've used only keys. I can either drop this lambda or enhance it so that it's identical if needed.

**Comment by cunningt** [comment]
Should we target this for 7-x?

**Comment by grgrzybek** [comment]
@cunningt I remember about this and I'll merge this for 7.9, but didn't check this yet ;)

File: modules/fuse-patch/patch-core/src/main/java/org/jboss/fuse/patch/impl/PatchServiceImpl.java [hunk 1]
```diff
@@ -1619,14 +1613,10 @@ private boolean refreshAutoDeployCache(File[] zips) {
 
     @Override
     public void undeploy(Patch patch) {
-        for (Iterator<DeployedPatch> iterator = deployedPatches.values().iterator(); iterator.hasNext(); ) {
-            DeployedPatch dp = iterator.next();
-            if (dp.getPatchData().getId().equals(patch.getPatchData().getId())) {
-                iterator.remove();
```
Comment (grgrzybek) [inline modules/fuse-patch/patch-core/src/main/java/org/jboss/fuse/patch/impl/PatchServiceImpl.java:1625]: @VratislavHais I had to revert this particular change - I needed this `iterator.remove()`.

**Comment by grgrzybek** [comment]
But besides that iterator line, I merged your PR - thanks!

**Comment by grgrzybek** [comment]
Merged with #371

**Pull request closed without merging**